### PR TITLE
Replaced ClassData with TheoryData

### LINKED ISSUE #17: Convert ClassData to TheoryData
TheoryData allows type checks during compile time because data types of test specimens are explict; this is preferred than the objects that ClassData returns.

### PR DESCRIPTION
**Pull request by itiannerona**
Resolves #17 . All tests pass.

### CODE REVIEW

**Comment by itiannerona** [comment]
I saw that Account is using the enum NormalBalance, where do you plan to use it?

File: source/Pacioli/Pacioli.Lib/Models/JournalEntry.cs [hunk 1]
```diff
@@ -29,7 +29,7 @@ public record JournalEntry
             EnsureArgNotEmpty(debits, nameof(debits), "Debits should not be empty");
             EnsureArgNotEmpty(credits, nameof(credits), "Credits should not be empty");
 
-            bool hasCommonAccount = debits.Any(dr => credits.Any(cr => dr.Account.Equals(cr.Account)));
+            bool hasCommonAccount = debits.Any(dr => credits.Any(cr => dr.Account.Name.Equals(cr.Account.Name)));
```
Comment (maratmuginov) [inline source/Pacioli/Pacioli.Lib/Models/JournalEntry.cs]: Why is only one property being compared now?

**Review by maratmuginov — COMMENTED** [review]
Why did you change the equality comparison on Account?

File: source/Pacioli/Pacioli.Lib/Models/JournalEntry.cs (hunk 1 above)
Comment (itiannerona) [reply source/Pacioli/Pacioli.Lib/Models/JournalEntry.cs]: I found the difference. I set both test data as credit and debit, where it should have been debit and debit or credit and credit.

**Review by maratmuginov — APPROVED** [review]

**Pull request merged**